src/openforms/api/utils.py

```diff
@@ -46,7 +46,12 @@ def get_model_serializer_instance(serializer: ModelSerializer) -> models.Model:
     return serializer.Meta.model()
 
 
-def mark_experimental(func_or_class):
+T = TypeVar(
+    "T", bound=type
+)  # ensure `mark_experimental` does not change the type signature
+
+
+def mark_experimental(func_or_class: T) -> T:
```

Comment:
```suggestion
def mark_experimental[T: type](func_or_class: T) -> T:
```

the [new syntax since 3.12](https://docs.python.org/3.12/reference/compound_stmts.html#type-params) makes this much less verbose